src/lib/span.ts

```diff
@@ -0,0 +1,38 @@
+export class Span {
+    spanId: string;
+    name: string;
+    startTime: number;
+    endTime: number | null;
+    parentSpanId: string | null;
+    children: Span[];
```

Comment:
Properties of classes should have an explicit `public` or `protected` specified. These are all public so use `public spanId: string` etc etc.

Also if they're meant to be private then always use the `protected` keyword unless they're meant to be extended in some way.